### PR TITLE
chunking vm requests

### PR DESCRIPTION
**Pull request by richardtief**
I have received feedback that virtual machines are missing. After a few tests I was able to confirm this and fix it. It looks like the amount of resource requests is limited to `10000`.

### CODE REVIEW

File: tools/Vrops.py [hunk 1]
```diff
@@ -142,7 +142,12 @@ def get_hosts(self, target, token, parent_uuids):
         return self.get_resources(target, token, parent_uuids, resourcekinds=["HostSystem"])
 
     def get_vms(self, target, token, parent_uuids):
-        return self.get_resources(target, token, parent_uuids, resourcekinds=["VirtualMachine"], data_receiving=True)
+        uuids_chunked = list(chunk_list(parent_uuids, 150))
```
Comment (viennaa) [inline tools/Vrops.py:145]: Why 150? Shouldn't they be chunked at 1000?

Comment (richardtief) [reply tools/Vrops.py:145]: We decreasing the amount of parent uuids here. Hosts to be precise.

File: tools/Vrops.py [hunk 2]
```diff
@@ -141,8 +141,25 @@ def get_cluster_and_datastores(self, target, token, parent_uuids):
     def get_hosts(self, target, token, parent_uuids):
         return self.get_resources(target, token, parent_uuids, resourcekinds=["HostSystem"])
 
-    def get_vms(self, target, token, parent_uuids):
-        return self.get_resources(target, token, parent_uuids, resourcekinds=["VirtualMachine"], data_receiving=True)
+    def get_vms(self, target, token, parent_uuids, vcenter_uuid):
+        amount_vms, api_responding, response_time = self.get_latest_stats_multiple(target, token, [vcenter_uuid],
```
Comment (viennaa) [inline tools/Vrops.py:145]:
```suggestion
        amount_vms, api_responding, _ = self.get_latest_stats_multiple(target, token, [vcenter_uuid],
```
```suggestion
        amount_vms, api_responding, response_time = self.get_latest_stats_multiple(target, token, [vcenter_uuid],
```

File: tools/Vrops.py [hunk 3]
```diff
@@ -141,8 +141,25 @@ def get_cluster_and_datastores(self, target, token, parent_uuids):
...
+                                                                                   ['summary|total_number_vms'],
+                                                                                   'Inventory')
+        number_of_vms = amount_vms[0].get('stat-list', {}).get('stat', [])[0].get('data', [0])[0] if \
+            api_responding == 200 else 0
+
+        # vrops cannot handle more than 10000 resource requests
+        if number_of_vms > 10000:
+            uuids_chunked = list(chunk_list(parent_uuids, int(round(len(parent_uuids) / 2, 0))))
```
Comment (viennaa) [inline tools/Vrops.py:153]: So basically we are chunking every 2 hosts? Am I getting this right?

Comment (richardtief) [reply tools/Vrops.py:153]: We making two chunks with the half of the hosts each.

Comment (richardtief) [reply tools/Vrops.py:153]: We assuming a uniform distribution over the hosts, yes.

Comment (viennaa) [reply tools/Vrops.py:153]: Can we make it two tiers?

```
> 10000 -> 2
> 20000 -> 4
```

Comment (viennaa) [reply tools/Vrops.py:153]: or better even: every 10000 more we multiply chunks by two.

Comment (richardtief) [reply tools/Vrops.py:153]: Ok, you are right. I think also we exceeding 20000 VMs in the near future. I adjust this.

File: tools/Vrops.py [hunk 4]
```diff
@@ -141,8 +141,27 @@ def get_cluster_and_datastores(self, target, token, parent_uuids):
...
-        return self.get_resources(target, token, parent_uuids, resourcekinds=["VirtualMachine"], data_receiving=True)
+    def get_vms(self, target, token, parent_uuids, vcenter_uuid):
+        amount_vms, api_responding, _ = self.get_latest_stats_multiple(target, token, [vcenter_uuid],
+                                                                       ['summary|total_number_vms'],
+                                                                       'Inventory')
+
+        number_of_vms = amount_vms[0].get('stat-list', {}).get('stat', [])[0].get('data', [0])[0] if \
+            api_responding == 200 else 0
+
+        # vrops cannot handle more than 10000 resource requests
```
Comment (viennaa) [inline tools/Vrops.py:152]:
```suggestion
        # vrops cannot handle more than 10000 uuids in a single request
```

**Review by viennaa — COMMENTED** [review]
`dynamic split factor` mmyd 👍 

please check the inline comment

**Review by viennaa — APPROVED** [review]
Thanks for the implementation and finding this bug in the first place. Like it.

**Pull request merged**